packages/studioCMS/src/integrations/studioCMSDashboard/routes/databaseSetup/main.astro

```diff
@@ -0,0 +1,190 @@
+              </sl-input>
+            </sl-details>
+
+            <sl-details id="oauthadmin" open>
+              <div class="textlabelinput" slot="summary"><sl-icon name="person-rolodex"></sl-icon> oAuth Admin Setup</div>
+              <sl-input 
+              name="oauth-admin-name"
+              id="oauth-admin-name"
+              placeholder="Username of Primary Admin (ex. GithubUsername)">
+                <div class="textlabelinput" slot="label">Default Site Admin <i>( Just set one, You can add more later )</i></div>
+              </sl-input>
+            </sl-details><br />
+
+
+            <div style="text-align:center">
+              <sl-button variant="success" type="submit" outline>Initialize Database!<div slot="suffix" class="i-mdi:login"></div></sl-button>
+            </div>
+
+          </form>
+
+          </div>
+        </div> 
+
+      <sl-alert id="form-error" variant="danger" duration="4000" closable>
+        <sl-icon slot="icon" name="exclamation-octagon"></sl-icon>
+        <strong>Something went wrong!</strong><br />
+        <div id = "form-error-message"></div>
+      </sl-alert>
+      <sl-alert id="form-success" variant="success" duration="4000" closable>
+        <sl-icon slot="icon" name="check2-circle"></sl-icon>
+        <strong>Success!</strong><br />
+        <div id="form-success-message"></div>
+      </sl-alert>
+
+
+  </div>
+</Layout>
+
+<script>
+	const alertElement = document.getElementById("form-error")!;
+  const errorMessageElement = document.getElementById("form-error-message")!;
+	const successAlertElement = document.getElementById("form-success")!;
+  const successMessageElement = document.getElementById("form-success-message")!;
+  const checkbox = document.getElementById('local-setup')!;
+  const localAdmin = document.getElementById('local-admin')!;
+  const oauthAdmin = document.getElementById('oauth-admin')!;
+
+  // Update the disabled state of the local-admin and oauth-admin details based on the checkbox state
+  checkbox.addEventListener('sl-change', () => {
+      //@ts-expect-error - checkbox comes from sl-checkbox from `Shoelace.syle`
+    if (checkbox.checked) {
+      localAdmin.removeAttribute("disabled")
+      //@ts-expect-error - show is not a standard method on HTMLElement it comes from `Shoelace.syle`
+      localAdmin.show();
+      //@ts-expect-error - hide is not a standard method on HTMLElement it comes from `Shoelace.syle`
+      oauthAdmin.hide();
+      oauthAdmin.setAttribute("disabled", "true")
+    } else {
+      //@ts-expect-error - hide is not a standard method on HTMLElement it comes from `Shoelace.syle`
+      localAdmin.hide();
+      localAdmin.setAttribute("disabled", "true")
+      oauthAdmin.removeAttribute("disabled")
+      //@ts-expect-error - show is not a standard method on HTMLElement it comes from `Shoelace.syle`
+      oauthAdmin.show();
+    }
+  });
+
+	document.forms[0]?.addEventListener("submit", async (e) => {
```

Comment:
In this case, atleast for now i think it should be fine, since this is only the First time setup page, and primarily used only to setup the tables with some dummy data to get started as well as the permission setup